src/lock.rs

```diff
@@ -1,147 +1,112 @@
-///
-/// line is:
-///   git@foo.com:bar/baz@<old>#egg=bar
-/// we want:
-///   git@foo.com:bar/baz@<new>@egg=bar
-///
-/// Return None if the name was not found, or Some(new_line)
-fn git_bump(line: &str, name: &str, git_ref: &str) -> Result<Option<String>, ParseError> {
-    if !line.contains('@') {
-        return Ok(None);
-    }
-    let chunks: Vec<_> = line.rsplit('@').collect();
-    // chunks is [git, foo:com:bar/baz, abce64#egg=bar]
-    let after_at = chunks.first().unwrap();
-    let chunks: Vec<_> = after_at.split('#').collect();
-    // chunks is [abce64, egg=bar]
-    if chunks.len() != 2 {
-        return Err(ParseError::new(&format!(
-            "expecting `<ref>#egg=<name>` after `@`, got '{}'",
-            after_at
-        )));
-    }
-    let dep_ref = chunks[0];
-
-    let start = line.len() - after_at.len();
-    let end = start + dep_ref.len();
-
-    let with_egg = chunks[1];
-    if !with_egg.starts_with("egg=") {
-        return Err(ParseError::new(&format!(
-            "expecting '{}' to start with `egg=`",
-            with_egg
-        )));
-    }
-    let dep_name = &with_egg[4..];
-    if dep_name != name {
-        return Ok(None);
-    }
-
-    let mut res = String::new();
-    res.push_str(&line[0..start]);
-    res.push_str(git_ref);
-    res.push_str(&line[end..]);
-    Ok(Some(res))
+struct GitBumper {
+    git_ref: String,
 }
 
-/// Bump the version number for a simple dependency specification
-///
-/// line is:
-///    foo==<old>
-///
-///  we want:
-///    foo==<new>
-///
-/// Return None if the name was not found, or Some(new_line)
-fn simple_bump(line: &str, name: &str, version: &str) -> Result<Option<String>, ParseError> {
-    if !line.contains("==") {
-        return Ok(None);
-    }
-    let words: Vec<_> = line.split("==").collect();
-    if words.len() != 2 {
-        return Err(ParseError::new(&format!(
-            "expecting `<name>==<version>`, got '{}'",
-            line
-        )));
-    }
-
-    let dep_name = words[0];
-    if dep_name != name {
-        return Ok(None);
-    }
-
-    Ok(Some(format!("{}=={}", dep_name, version)))
+impl GitBumper {
+    fn new(git_ref: &str) -> Self {
+        GitBumper {
+            git_ref: git_ref.to_string(),
+        }
+    }
+}
+
+impl Bumper for GitBumper {
+    fn bump(&self, dep: &mut LockedDependency) -> bool {
+        if let LockedDependency::Git(g) = dep {
+            g.bump(&self.git_ref)
+        } else {
+            false
+        }
+    }
 }
 
 impl Lock {
-    pub fn new(contents: &str) -> Lock {
-        Lock {
-            contents: contents.to_owned(),
+    pub fn from_string(string: &str) -> Result<Self, Error> {
+        let mut dependencies = vec![];
+        for (i, line) in string.lines().enumerate() {
+            let line = line.trim();
+            if line.is_empty() {
+                continue;
+            }
+            if line.starts_with('#') {
+                continue;
+            }
```

Comment:
Hmm I might put those two in the same if as A || B, but that's pretty subjective.

Reply:
It's a bit clearer